base/range.jl

```diff
@@ -67,6 +67,8 @@ immutable UnitRange{T<:Real} <: OrdinalRange{T,Int}
 end
 UnitRange{T<:Real}(start::T, stop::T) = UnitRange{T}(start, stop)
 
+colon() = Colon()
```

Comment:
I don't think we need this. `colon` is never called with 0 arguments as far as I know. The front end can just produce `(top :)` or `(call (top Colon))` instead.